src/Symfony/Component/Clock/ClockAwareTrait.php

```diff
@@ -29,8 +29,13 @@ public function setClock(ClockInterface $clock): void
         $this->clock = $clock;
     }
 
+    /**
+     * @return Moment
+     */
     protected function now(): \DateTimeImmutable
     {
-        return ($this->clock ??= new Clock())->now();
+        $now = ($this->clock ??= new Clock())->now();
+
+        return $now instanceof Moment ? $now : Moment::createFromInterface($now);
```

Comment:
Would it make sense to move this ternary logic into `Moment::createFromInterface()`?

Reply:
Not sure, I'd prefer saving the method call as early as possible (the internal implementation should already implement this logic so adding it on our side wouldn't be needed)